r/tests/testthat/test-dplyr-distinct.R

```diff
@@ -0,0 +1,73 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+skip_if_not_available("dataset")
+
+library(dplyr)
+library(stringr)
+
+tbl <- example_data
+tbl$some_grouping <- rep(c(1, 2), 5)
+
+test_that("distinct()", {
+  expect_dplyr_equal(
+    input %>%
+      distinct(some_grouping, lgl) %>%
+      collect(),
+    tbl
+  )
+})
+
+test_that("distinct() works without any variables", {
+  expect_dplyr_equal(
+    input %>%
+      distinct() %>%
+      collect(),
+    tbl
+  )
+})
+
+test_that("distinct() can retain groups", {
+  skip("ARROW-13777 - internally uses mutate on grouped data; should work after this is merged")
```

Comment:
I think you can remove your mutate call though and these should pass if you let group_by do the work

Reply:
Will try that

Reply:
I also now have a PR open for ARROW-13777

Reply:
Update: that PR has now merged, so you can rebase and unskip